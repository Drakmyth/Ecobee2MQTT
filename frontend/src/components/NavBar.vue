<script setup lang="ts">
import { RouterLink } from 'vue-router';
</script>

<template>
    <nav class="navbar">
        <span class="navbar-title">Ecobee2MQTT</span>
        <ul>
            <li>
                <RouterLink to="/">Devices</RouterLink>
            </li>
            <li>
                <RouterLink to="/settings">Settings</RouterLink>
            </li>
        </ul>
    </nav>
</template>

<style>
.navbar {
    display: flex;
    flex-direction: row;
}

.navbar .navbar-title {
    align-self: center;
}

.navbar ul {
    list-style-type: none;
    display: flex;
    flex-direction: row;
    --nav-gap: 12px;
    margin: 0 var(--nav-gap);
    padding: 0;
    column-gap: var(--nav-gap);
}

.navbar a {
    padding: 0.5rem;
    display: block;
}
</style>
