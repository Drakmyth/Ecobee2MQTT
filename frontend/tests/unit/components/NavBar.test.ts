import { shallowMount } from '@vue/test-utils';
import NavBar from '@/components/NavBar.vue';

describe('NavBar.vue', () => {
    test('renders correctly', () => {
        const wrapper = shallowMount(NavBar);
        expect(wrapper.html()).toMatchSnapshot();
    });
});
