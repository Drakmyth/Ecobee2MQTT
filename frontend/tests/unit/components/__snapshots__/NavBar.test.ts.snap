// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar.vue renders correctly 1`] = `
"<nav class=\\"navbar\\"><span class=\\"navbar-title\\">Ecobee2MQTT</span>
  <ul>
    <li>
      <router-link-stub to=\\"/\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
    </li>
    <li>
      <router-link-stub to=\\"/settings\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
    </li>
  </ul>
</nav>"
`;
